src/main/java/mods/railcraft/common/blocks/wayobjects/RoutingLogic.java

```diff
@@ -133,6 +134,17 @@ private Condition parseLine(String line, Deque<Condition> stack) throws RoutingL
         } catch (Exception ex) {
             throw new RoutingLogicException("railcraft.gui.routing.logic.malformed.syntax", line);
         }
+        if (line.equals("TRUE"))
+            return new ConstantCondition(true);
+        if (line.equals("FALSE"))
+            return new ConstantCondition(false);
+        try {
+            return new ConstantCondition(Integer.parseInt(line));
+        } catch (NumberFormatException ignored) {
+            // not an integer; pass through
+        } catch (IllegalArgumentException ex) {
+            throw new RoutingLogicException("railcraft.gui.routing.logic.constant.invalid", line);
+        }
```

Comment:
This feels fundamentally flawed. How do you turn a ConstantCondition into a return statement?

As written you'd have to do:
```
AND
Dest=Town
13
```
...to return 13. Which does not feel very intuitive and does not match the original issues description.